### DISCUSSION TITLE
[ Projects Beta ] too easy to delete fields

### DISCUSSION THREAD

**Discussion by chrisroedig** [community/community#13373] [Projects and Issues]
![image](https://user-images.githubusercontent.com/730275/159786785-539957e9-22b1-476c-94bf-53ad80216ff2.png)

It's way too easy to accidentally delete a field from the table. I meant to just not show the column, but I accidentally deleted the sprint from all the issues.

It might be better to not have that menu option and only allow users to delete fields from the admin page.
Another option would be to throw in a confirmation prompt.

**Comment by raihle** [comment]
Seems "answer" is all I can do in a discussion... this is not an answer, but another facet of the same issue.

Conversely, I couldn't figure out how to delete fields until I found this post. Add field administration to a project settings page, and remove "dangerous" options from the boards!

**Comment by danicahelb** [comment]
At least give us the option to undo the deletion of a field!

**Comment by tay1orjones** [comment]
This is a big problem. My team is having to re-apply a custom field to ~500 issues after the field was accidentally deleted. 

It was a single select "area" of category-like buckets we organize our issues by. We have recreated the field but there is little way of knowing what they were previously categorized as.

Another significant side effect is any view/tab that was using that field as a "grouped by" or similar no longer has this set up. We have to go fix all our views now as well.

---

Echoing others in this discussion:

- Remove destructive actions from the board
  - Only allow them in the settings pane
  - Prompt a dialog and list all the views/tabs that would be impacted by the deletion
- Add a highlighted "danger" section to the field settings
- Add a way to restore deleted fields
- Provide more granular control over who can edit project settings
  - Most team members only need access to change field value on an issue, or make a new view/tab, etc. not project-level settings actions.

**Comment by DNin01** [comment]
Apparently, after an update this button is now labeled _Delete field **dot dot dot**_.
![Dropdown menu containing Delete field… button](https://github.com/community/community/assets/106490990/27190cf2-e0e3-4d83-bdaf-4db86378e079)

Still not going to try it but from the ellipsis, it looks like it would open a modal of some sort, so there's now a second step.

Regardless, I never use that button; I'd rather delete fields through Settings as it's still somewhat destructive to be on the dropdown menu, but now that it supposedly has a second step, I don't really have a strong opinion now.

**Reply by DNin01** [reply]
Now look what it says!
![Field dropdown menu with "Field settings..." option](https://github.com/community/community/assets/106490990/a4bce153-c7a5-414b-9bfe-2406738008d6)

Instead of having a delete button right there in the dropdown menu, you have to go to the settings for the field before you can delete it. That's a much better place for that option.

**Comment by TheJagpreet** [comment]
I faced a similar issue, where instead of deleting `Sprint5`, I deleted `Sprint` and lost all progress. Is there a way, that can be reverted back?

**Comment by javedali-dev** [comment]
Even I made the same mistake that @jagpreetsinghsasan did. and now i lost all the progress. Github should provide some solution for this.
